cartofacade/cgo_api.go

```diff
@@ -0,0 +1,401 @@
+func (vc *Carto) TerminateCarto(ctx context.Context) error {
+	status := C.viam_carto_terminate(&vc.value)
+
+	if err := getErrorFromStatusCode(status); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// AddSensorReading is a wrapper for viam_carto_add_sensor_reading
+func (vc *Carto) AddSensorReading(ctx context.Context, readings []byte, timestamp time.Time) error {
+	sensorReading := C.viam_carto_sensor_reading{}
+	sensorReading.sensor_reading = C.blk2bstr(unsafe.Pointer(&readings[0]), C.int(len(readings)))
+
+	sensorReading.sensor_reading_time_unix_micro = C.ulonglong(timestamp.UnixMicro())
+
+	resp := AddSensorReading{value: sensorReading}
+
+	status := C.viam_carto_add_sensor_reading(vc.value, &resp.value)
+
+	if err := getErrorFromStatusCode(status); err != nil {
+		return err
+	}
+
+	status = C.viam_carto_add_sensor_reading_destroy(&resp.value)
+	if err := getErrorFromStatusCode(status); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetPosition is a wrapper for viam_carto_get_position
+func (vc *Carto) GetPosition(ctx context.Context) (GetPositionHolder, error) {
+	resp := GetPositionResponse{value: C.viam_carto_get_position_response{}}
+
+	status := C.viam_carto_get_position(vc.value, &resp.value)
+
+	if err := getErrorFromStatusCode(status); err != nil {
+		return GetPositionHolder{}, err
+	}
+
+	getPositionHolder := GetPositionHolder{
+		x: float64(resp.value.x),
+		y: float64(resp.value.y),
+		z: float64(resp.value.z),
+
+		ox:    float64(resp.value.o_x),
+		oy:    float64(resp.value.o_y),
+		oz:    float64(resp.value.o_z),
+		theta: float64(resp.value.theta),
+
+		real: float64(resp.value.real),
+		imag: float64(resp.value.imag),
+		jmag: float64(resp.value.jmag),
+		kmag: float64(resp.value.kmag),
+
+		compReference: bstringToGoString(resp.value.component_reference),
+	}
+
+	status = C.viam_carto_get_position_response_destroy(&resp.value)
+	if err := getErrorFromStatusCode(status); err != nil {
+		return GetPositionHolder{}, err
+	}
+
+	return getPositionHolder, nil
+}
+
+// GetPointCloudMapResponse is a struct to hold the c type viam_carto_get_point_cloud_map_response
+type GetPointCloudMapResponse struct {
+	value C.viam_carto_get_point_cloud_map_response
+}
+
+// GetPointCloudMap is a wrapper for viam_carto_get_point_cloud_map
+func (vc *Carto) GetPointCloudMap(ctx context.Context) ([]byte, error) {
+	// TODO: determine whether or not return needs to be a pointer for performance reasons
+	resp := GetPointCloudMapResponse{value: C.viam_carto_get_point_cloud_map_response{}}
+
+	status := C.viam_carto_get_point_cloud_map(vc.value, &resp.value)
+
+	if err := getErrorFromStatusCode(status); err != nil {
+		return nil, err
+	}
+
+	status = C.viam_carto_get_point_cloud_map_response_destroy(&resp.value)
+	if err := getErrorFromStatusCode(status); err != nil {
+		return nil, err
+	}
+
+	bytes := (*[]byte)(unsafe.Pointer(&resp.value.point_cloud_pcd))
```

Comment:
I don' think this is correct... `point_cloud_pcd` is a `bstring`, shouldn't we be converting the better string into a go byte slice? Where did you find docs on how to do this?